src/PwrDrvr.LambdaDispatch.Router/LambdaInstanceManager.cs

```diff
@@ -307,20 +352,31 @@ private async Task ManageCapacity()
                 StartNewInstance();
               }
             }
-            else if (!deferredScaleInNewDesiredInstanceCount.HasValue)
+            else if (!deferredScaleInNewDesiredInstanceCount.HasValue
+                    && newDesiredInstanceCount < _desiredInstanceCount)
             {
-              // Schedule a scale down
+              _logger.LogDebug("ManageCapacity - Scheduling scale in: _desiredInstanceCount {_desiredInstanceCount} -> {newDesiredInstanceCount}, ewmaScalerDesiredInstanceCount {ewmaScalerDesiredInstanceCount}, simpleScalerDesiredInstanceCount {simpleScalerDesiredInstanceCount}, prevDesiredInstanceCount {prevDesiredInstanceCount}, requestsPerSecondEWMA {requestsPerSecondEWMA}, requestDurationEWMA {requestDurationEWMA}, _runningInstanceCount {}, _startingInstanceCount {}, _stoppingInstanceCount {}",
+                _desiredInstanceCount, newDesiredInstanceCount, ewmaScalerDesiredInstanceCount != null ? ewmaScalerDesiredInstanceCount.Value : null, simpleScalerDesiredInstanceCount, prevDesiredInstanceCount, Math.Round(requestsPerSecondEWMA, 1), Math.Round(requestDurationEWMA, 1), _runningInstanceCount, _startingInstanceCount, _stoppingInstanceCount);
+
+              // Schedule a scale in
               deferredScaleInNewDesiredInstanceCount = newDesiredInstanceCount;
-              cts = new CancellationTokenSource(TimeSpan.FromMilliseconds(500));
+              // This needs to be long enough for us to get a message from the background dispatcher
+              cts = new CancellationTokenSource(TimeSpan.FromMilliseconds(300));
             }
-            else if (deferredScaleInNewDesiredInstanceCount > newDesiredInstanceCount)
+            else if (deferredScaleInNewDesiredInstanceCount.GetValueOrDefault() > newDesiredInstanceCount)
```

Comment:
THIS WAS THE ISSUE!!!

- The scale down was broken in all cases *except* when there was a scale from load to no load
- When scaling to no load it would eventually scale to 0 due to the 5 second timer seeing no messages read from the channel
- But if it went from high load to low load this missing `.Value` or `.GetValueOrDefault()` would cause this block to get skipped and the next block would clear the pending scale down every time
- Once this is fixed, it triggers another issue which is that the scale-to-zero logic is too sensitive, which then causes frequent scales to zero which then causes the huge queue delays, etc.